Resolve #3777 and #3779, issues of some front-end outputs

yuliu opened the pull request:
Resolve #3777, partially (see the note below):

Fix **empty post subject not well handled in warning & modcp pages** in (mostly for compatible issues):
- `./warnings.php?uid={$uid}`
- `./warnings.php?action=warn&uid={$uid}&pid={$pid}`
- `./warnings.php?action=view&wid={$wid}`
- `./modcp.php?action=modqueue&type=posts`
- `./modcp.php?action=modqueue&type=attachments`
- **`./modcp.php?action=modlogs`:** `pid` value in table `moderatorlog` is not correctly stored for some mod actions on posts, can't really depend on that.
- `./modcp.php?action=ipsearch&ipaddress={$ip}&search_posts=1` _(added in a recent commit)_
- `./usercp.php`
- `./usercp.php?action=attachments`
- `./usercp.php?action=drafts`
- `./showthread.php?mode=threaded&tid={$tid}&pid={$pid}` _(added in a recent commit)_

Fix **empty template variables**: `$timeremaining` for displaying banned time on member profile.

Fix **file size unit in Alert box text for attachment size over PHP limit not inconsistent with language file**. It's a quick fix and it's somehow unfriendly to end-user. Hope anyone experienced with JavaScript will code a function similar to `get_friendly_size()` to display the value.

And fix #3779 hard-coded CSS stylesheet id when exporting PMs as HTML.

**Note:**

1. As for the inconsistent language usages of text for navigation breadcrumb on UserCP page, changes upon that might lead to more trouble for current users. So I leave it untouched. If we have to deal with this minor problem, better do it in feature major version upgrade.
1. I'd like to suggest changes for non-well-formatted table's `class` attribute output in AdminCP in another PR, since this one focuses on the front-end.
1. Since a literal string `RE: ` is hardcoded in current MyBB deeply, it's better to address its problem in a separated issue.

yuliu committed:
Fix remaining ban time not showing in member profile (in mybb:#3777).

yuliu committed:
Quick fix inconsistent allowed size unit for `php_max_upload_size` in JavaScript and language entry (in mybb:#3777).

yuliu committed:
Fix incorrect displaying for warning by posts with empty subject (in mybb:#3777).

yuliu committed:
Fix incorrect display for posts with empty subject in UserCP section (in mybb:#3777).

yuliu committed:
Fix incorrect display for posts with empty subject in ModCP section (in mybb:#3777).

yuliu committed:
Replace hard-coded stylesheet query id with user-set theme id, or default/global template theme id if lacking one.

Eldenroot commented:
Good job keep it

yuliu committed:
Increse version to 1822 for `post.js` and `post_javascript` template.

euantorano commented on modcp.php:
```diff
@@ -2452,6 +2460,10 @@
 			$attachment['filesize'] = get_friendly_size($attachment['filesize']);
+			if(empty($attachment['postsubject']))
+			{
+				$attachment['postsubject'] = "RE: ".$attachment['threadsubject'];
```
This should probably be translatable via a language variable.

yuliu replied:
The `RE:` prefix for thread replies is indeed hardcoded deeply in current MyBB.

I'll only try to update those variables in this PR with a translatable entry.

euantorano replied:
I feared that may be the case. In that case, just leave the ones in this PR for now, and we'll open another issue to try and patch them all at once.

euantorano commented on usercp.php:
```diff
@@ -3855,6 +3859,10 @@
 			$attachment['threadsubject'] = htmlspecialchars_uni($parser->parse_badwords($attachment['threadsubject']));
+			if(empty($attachment['subject']))
+			{
+				$attachment['subject'] = "RE: ".$attachment['threadsubject'];
```
Again, this should probably be translatable.

euantorano commented on usercp.php:
```diff
@@ -3317,6 +3317,10 @@
 
+			if(empty($draft['subject']))
+			{
+				$draft['subject'] = "[".$lang->post_no_subject."]";
```
This might be nice to be translatable, since we're trying to minimise the amount of hardcoded HTML and other text.

yuliu replied:
I followed the way of handling empty post subject in `showthread.php`. Then I find that I should also make `[]` as translatable, you can see I haven't enclosed them with brackets in `modcp.php`.

I'll try to update them and `showthread.php` as well.

euantorano replied:
Ok. It seems we still have a lot of hardcoded stuff scattered about in that case 😢

yuliu replied:
Seems `no_subject` is only hardcoded in `showthread.php` before this PR. But for `RE: `... pretty a lot.

euantorano replied:
We'll definitely do a separate PR for `RE:` then, otherwise it would make following PRs nearly impossible.

euantorano commented on usercp.php:
```diff
@@ -4057,8 +4065,12 @@
 				{
+					if(empty($warning['post_subject']))
+					{
+						$warning['post_subject'] = "[".$lang->post_no_subject."]";
```
See above regarding translations.

euantorano commented on jscripts/post.js:
```diff
@@ -181,7 +181,7 @@ var Post = {
 		if (totalSize > php_max_upload_size && php_max_upload_size > 0)
 		{
-			var php_max_upload_size_pretty = Math.round(php_max_upload_size / 1e4) / 1e2;
+			var php_max_upload_size_pretty = php_max_upload_size;
```
We definitely need a friendly size for this, I can write one if you open this PR to allow modifications from team members - see here for instructions: https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

yuliu replied:
The `Allow edits from maintainers. ` option is already ticked. Should I change any other settings?

Now working on the `no subject` language entries.

euantorano replied:
Great, I'll work on a function to format file sizes for JS as part of this PR later on today.

yuliu replied:
Borrowed a function from https://stackoverflow.com/a/20732091/6681141

euantorano reviewed (changes requested):
Looks great, just some minor comments that are mostly nitpicks.

yuliu committed:
Replace hardcoded `no subject`, `RE: `(only in this PR) with translatable variables.
Replace hardcoded `no subject` in `showthread`'s threaded view to use `{$LANG_VAR_FOR_PREFIX}.{$thread['subject']}`.
Remove unused language entries for `no subject` from those added formerly in this PR and `showthread` page.

yuliu reviewed (commented):
Forgot to mention the empty subject when searching IP in posts is also replaced with the language variable in this commit. Will add this information in the main PR description.

Eldenroot commented:
Ok, I didnt know that... Good job!

Eldenroot commented:
I have applied to my board :) more testers needed 👍

euantorano commented:
See my above comment about friendly size - I will write the code for this this weekend and add it to this PR.

yuliu committed:
Update to use a human friendly size for uploads over php_max_upload_size.
Bump template version to 1823.
Tune some language entries.

yuliu committed:
Merge branch 'feature' into feature-fix-3777

yuliu committed:
Merge branch 'feature' into feature-fix-3777

yuliu commented:
> See my above comment about friendly size - I will write the code for this this weekend and add it to this PR.

Have borrowed one from stackoverflow which seems good, and size units are backed by language entries.

Ben-MyBB commented:
@yuliu
